datagateway_api/src/search_api/models.py

```diff
@@ -43,102 +150,130 @@ class Dataset(PaNOSCAttribute):
     and Technique
     """
 
+    _related_fields_with_min_cardinality_one: ClassVar[List[str]] = [
+        "documents",
+        "techniques",
+    ]
     _text_operator_fields: ClassVar[List[str]] = ["title"]
 
-    pid: StrictStr
-    title: StrictStr
-    is_public: StrictBool = Field(alias="isPublic")
+    pid: str
+    title: str
+    is_public: bool = Field(alias="isPublic")
     creation_date: datetime = Field(alias="creationDate")
-    size: Optional[StrictInt]
+    size: Optional[int] = None
+
+    documents: List["Document"] = []
+    techniques: List["Technique"] = []
+    instrument: Optional["Instrument"] = None
+    files: Optional[List["File"]] = []
+    parameters: Optional[List["Parameter"]] = []
+    samples: Optional[List["Sample"]] = []
+
+    @validator("is_public", pre=True, always=True)
+    def set_is_public(cls, value):  # noqa: B902, N805
+        if not value:
+            return value
 
-    documents: List["Document"]
-    techniques: List["Technique"]
-    instrument: Optional["Instrument"]
-    files: Optional[List["File"]]
-    parameters: Optional[List["Parameter"]]
-    samples: Optional[List["Sample"]]
+        creation_date = datetime.fromisoformat(value)
+        current_datetime = datetime.now(timezone.utc)
+        three_years_ago = current_datetime - relativedelta(years=3)
+        return creation_date < three_years_ago
```

Comment:
Thanks for sorting this out inside this PR, it's nice to see that it doesn't require some difficult logic. We should probably check the date checks do return True for ISIS data but that's a task at the end when we go through the code and work out what's left to resolve

Reply:
This is tested in the tests but I agree, we should check this for ISIS at the end.

Your comment has made me look at this code once again and I think it is better if the `years` value is a config value. The current code would work for ISIS but I don't really know how other facilities define open data. What do you think?

Reply:
Yeah that would probably be good to make that configurable. I'll create an issue for it and we can tackle it separately